src/Compilers/Core/Portable/Compilation/IImportScope.cs

```diff
@@ -0,0 +1,111 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Immutable;
+
+namespace Microsoft.CodeAnalysis
+{
+    /// <summary>
+    /// Represents the set of symbols that are imported to a particular position in a source file. Each import has a
+    /// reference to the location the import directive was declared at.  For the <see cref="IAliasSymbol"/> import, the
+    /// location can be found using either <see cref="ISymbol.Locations"/> or <see
+    /// cref="ISymbol.DeclaringSyntaxReferences"/> on the <see cref="IAliasSymbol"/> itself.  For <see cref="Imports"/>
+    /// or <see cref="XmlNamespaces"/> the location is found through <see
+    /// cref="ImportedNamespaceOrType.DeclaringSyntaxReference"/> or <see
+    /// cref="ImportedXmlNamespace.DeclaringSyntaxReference"/> respectively.
+    /// </summary>
+    /// <remarks>
+    /// <list type="bullet">
+    /// <item>Scopes returned will always have at least one non-empty property value in them.</item>
+    /// <item>Symbols may be imported, but may not necessarily be available at that location (for example, an alias
+    /// symbol hidden by another symbol).</item>
+    /// <item>
+    /// In C# there will be an <see cref="IImportScope"/> for every containing namespace-declarations that include any
+    /// import directives.  There will also be an <see cref="IImportScope"/> for the containing compilation-unit if it
+    /// includes any import directives or if there are global import directives pulled in from other files.
+    /// </item>
+    /// <item>
+    /// In Visual Basic there will only be at most two <see cref="IImportScope"/>s returned for any position.  There
+    /// will be a scope for the containing compilation unit if it includes any import directives.  There can also be a
+    /// scope representing any imports specified at the project level.
+    /// </item>
+    /// <item>
+    /// Elements of any property have no defined order.  Even if they represent items from a single document, they are
```

Comment:
>Elements of any property have no defined order. [](http://example.com/codeflow?start=8&length=47)

nit: This surprised me a bit. But on second thought, if providing the compiler order wouldn't help customers anyways, it's fine to not help that service ;-)